.golangci.yml

```diff
@@ -25,6 +25,7 @@ issues:
     - "`defaultRetryDuration` is unused" # used by APM and Process
     - "`defaultRetries` is unused"       # used by APM and Process
     - "python._Ctype_char, which can be annoying to use" # ignore warning about returning unexported field from CGO
+    - "return value of `config.BindEnv` is not checked" # ignore BindEnv error return not being checked
```

Comment:
I agree that the error can be ignored (only case when it can return an error is https://github.com/DataDog/viper/blob/8e7459bcf337c155af4be4d26f6d636f1204a1fb/viper.go#L965, which we can completely ignore).

That said, instead of this, I think we should modify the `Config` interface and `pkg/config/viper.go` to simply make `BindEnv` not return any error. WDYT?

Reply:
➕ I was going to comment the same thing

Reply:
I thought of this, but the official viper interface returns an error https://pkg.go.dev/github.com/DataDog/viper#BindEnv and I think we want to stay similar to it.

Reply:
I don't think we're aiming for the `Config` interface to match the `viper` lib exactly to be honest. Although I agree that it's better for our viper fork to have the same function signatures as upstream.